### PR TITLE
Katie and Katherine 

### PR DESCRIPTION
**Pull request by katietj**
# Adagrams
Congratulations! You're submitting your assignment.

## Comprehension Questions

Feature | Feedback
--- | ---
What are the components that make up a method? | The parts of a method are the signature, body, and what the method will return. 
What are the advantages of using git when collaboratively working on one code base? | Version control, multiple people can work and both make changes. 
What kind of relationship did you and your pair have with the unit tests? | Helpful validation that we were moving in the right direction with our code. 
Does your code use any methods from the `Enumerable` mixin? If so, where and why was it helpful? | We used .max and .map and they were helpful to make our code shorter and more readable. 
What was one method you and your pair used to debug code? | We used awesome print and IRB to work out some difficult code sections. We also used whiteboards to help us draw out what we wanted to do. 
What are two discussion points that you and your pair discussed when giving/receiving feedback from each other that you would be willing to share? | We discussed how we both were hesitant to ask for help from instructors, but when we did ask for help it really helped us guide our process. We talked about how it was easier on the last day to communicate with each other about code. Our verbal ruby language skills have really gotten better over the week!

### CODE REVIEW

File: adagrams.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
+#Katie and Katherine
```
Comment (CheezItMan) [inline adagrams.rb:1]: Why is this file dupplicated?

File: lib/adagrams.rb [hunk 2]
```diff
@@ -0,0 +1,151 @@
+#Katie and Katherine
+
+letters = %w(a a a a a a a a a b b c c d d d d e e e e e e e e e e e e f f g g g h h i i i i i i i i i k l l l l m m n n n n n n o o o o o o o o p p q r r r r r r s s s s t t t t t u u u u v v w w x y y z)
+users_hand = []
+
+def draw_letters
+  letters = %w(a a a a a a a a a b b c c d d d d e e e e e e e e e e e e f f g g g h h i i i i i i i i i k l l l l m m n n n n n n o o o o o o o o p p q r r r r r r s s s s t t t t t u u u u v v w w x y y z)
```
Comment (CheezItMan) [inline lib/adagrams.rb:7]: Good data structure, but why is it a duplicate of the variable above?

File: lib/adagrams.rb [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  end
+  10.times do |letters|
+    users_hand << letter_pool.delete_at(rand(letter_pool.length))
+  end
+  return users_hand
+end
+
+
+
+input = ""
```
Comment (CheezItMan) [inline lib/adagrams.rb:20]: Why these local variables?

File: adagrams.rb [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+end
+
+
+
+input = ""
+letters_in_hand = []
+
+def uses_available_letters?(input, letters_in_hand)
+  input =  input.split('').each {|c|}
+  input.each_with_index do |letter|
```
Comment (CheezItMan) [inline adagrams.rb:25]: With this method of the answer is `true` it instead returns the array.

File: adagrams.rb [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+      letters_in_hand.delete_at(letters_in_hand.find_index(letter))
+
+    elsif  letters_in_hand.include?(letter) == false
+
+      return false
+
+    end
+  end
+end
```
Comment (CheezItMan) [inline adagrams.rb:37]: Should have a `return true` before this `end`

File: adagrams.rb [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    scores << score_word(word)
+  end
+
+  words.each do |word|
+    lengths << word.length
+  end
+
+
+  i = 0
+  words.each do |key, value|
```
Comment (CheezItMan) [inline adagrams.rb:123]: Words is an array, why `key, value`?

File: adagrams.rb [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+  lowest_word_length = 0
+  highest_score = scores.max
+
+  word_scores.each do |word_hash|
+    if word_hash[:score] == highest_score
+      winner_array << word_hash
+    end
+  end
+
+  if winner_array.length < 1
```
Comment (CheezItMan) [inline adagrams.rb:139]: If there's less than 1 winner?  That means if there's 0 winners.

Also everything in here is much more complicated than it needs to be.

You could do:

```ruby
winning_word = winner_array.reduce(winner_array.first) do |current_max, current|
  if current[:word].length < current_max[:word].length && current_max[:word].length < 10
    current
  elsif current[:word].length == 10 && current_max[:word].length < 10
    current
  else
     current_max
  end
end

return winning_word[:word]
```

**Comment by CheezItMan** [comment]
# Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | Check, glad you both feel more comfortable asking questions
Both teammates contributed to the codebase |  Check, I assume
Small commits with meaningful commit messages | Pretty good, but instead of talking about waves describe the functionality your commits added in the commit messages.
**Code Requirements** |
`draw_letters` method |  Check
Uses appropriate data structure to store the letter distribution | Check, well done
All tests for `draw_letters` pass | Check
`uses_available_letters?` method | See my notes, small bug.
All tests for `uses_available_letters?` pass |  **ERROR** It's returning an array instead of true.
`score_word` method |  check
Uses appropriate data structure to store the letter scores |  Check
All tests for `score_word` pass |  Check
`highest_score_from` method | **DOES NOT WORK** See my notes
Appropriately handles edge cases for tie-breaking logic |
All tests for `highest_score_from` pass |  **ERROR** The winning word is not being selected!
**Overall** |  You had some issues with the tests and with `highest_score_from`.  See my notes and let me know if you have questions.  You did hit **most** of the learning goals, but need to review the failed tests and my inline notes.
